pkg/discovery/p2p/consensus_peer.go

```diff
@@ -0,0 +1,72 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/ethpandaops/ethcore/pkg/execution/mimicry"
+	"github.com/ethpandaops/xatu/pkg/proto/xatu"
+)
+
+type ConsensusPeer struct {
+	log logrus.FieldLogger
+
+	nodeRecord string
+
+	// TODO: replace with consensus ethcore client
+	client *mimicry.Client
+
+	handlerFunc func(ctx context.Context, status *xatu.ConsensusNodeStatus)
+}
+
+func NewConsensusPeer(ctx context.Context, log logrus.FieldLogger, nodeRecord string, handlerFunc func(ctx context.Context, status *xatu.ConsensusNodeStatus)) (*ConsensusPeer, error) {
+	client, err := mimicry.New(ctx, log, nodeRecord, "xatu")
+	if err != nil {
+		return nil, err
+	}
+
+	return &ConsensusPeer{
+		log:         log.WithField("node_record", nodeRecord),
+		nodeRecord:  nodeRecord,
+		client:      client,
+		handlerFunc: handlerFunc,
+	}, nil
+}
+
+func (p *ConsensusPeer) Start(ctx context.Context) (<-chan error, error) {
+	response := make(chan error, 1)
+
+	p.client.OnHello(ctx, func(ctx context.Context, hello *mimicry.Hello) error {
```

Comment:
just duped the execution ethcore stuff here, this would use w/e we do for the ethcore consensus stuff